src/test/java/mmosii/bookstore/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,257 @@
+package mmosii.bookstore.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import mmosii.bookstore.dto.shoppingcart.CartItemDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemRequestDto;
+import mmosii.bookstore.dto.shoppingcart.CartItemUpdateDto;
+import mmosii.bookstore.dto.shoppingcart.ShoppingCartDto;
+import mmosii.bookstore.mapper.CartItemMapper;
+import mmosii.bookstore.mapper.ShoppingCartMapper;
+import mmosii.bookstore.model.Book;
+import mmosii.bookstore.model.CartItem;
+import mmosii.bookstore.model.ShoppingCart;
+import mmosii.bookstore.model.User;
+import mmosii.bookstore.repository.book.BookRepository;
+import mmosii.bookstore.repository.shoppingcart.CartItemRepository;
+import mmosii.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mmosii.bookstore.repository.user.UserRepository;
+import mmosii.bookstore.service.impl.ShoppingCartServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@ExtendWith(MockitoExtension.class)
+public class ShoppingCartServiceTest {
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
```

Comment:
You can try don't mock mappers and make them @Spy. 
```
@Spy
private ShoppingCartMapper shoppingCartMapper = new ShoppingCartMapper Impl();
```
It decreases your amount of code.

Reply:
@mmosii pls address Maksym's comment - `@Spy` will reduce the amount of boilerplate in your tests